client/src/main/java/com/battleships/start_window/data_insertion/SettingsTextData.java

```diff
@@ -0,0 +1,80 @@
+package com.battleships.start_window.data_insertion;
+
+import com.battleships.LogMessages;
+import com.battleships.Translator;
+import com.battleships.start_window.connection.Connection;
+import com.battleships.start_window.connection.ConnectionInfo;
+import javafx.beans.binding.Bindings;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Optional;
+
+public class SettingsTextData {
+    @FXML
+    private Button connectToServerButton;
+    @FXML
+    private Button disconnectFromServerButton;
+    @FXML
+    private TextField nameTextField;
+    @FXML
+    private TextField ipTextField;
+    private final static Logger logger = LogManager.getLogger(SettingsTextData.class);
+
+    @FXML
+    public void initialize() {
+        bindTextFieldsWithTranslation();
+        bindConnectToServerButton();
+        setOnActionToButtons();
+    }
+
+    private void bindTextFieldsWithTranslation() {
+        Translator.bind(connectToServerButton.textProperty(), "connect");
+        Translator.bind(disconnectFromServerButton.textProperty(), "disconnect");
+        Translator.bind(nameTextField.promptTextProperty(), "player_name");
+    }
+
+    private void bindConnectToServerButton() {
+        connectToServerButton.disableProperty().bind(Bindings.isEmpty(ipTextField.textProperty()));
+    }
+
+    private void setOnActionToButtons() {
+        connectToServerButton.setOnAction(e -> connectToServerButtonAction());
+        disconnectFromServerButton.setOnAction(e -> Connection.INSTANCE.disconnect());
+    }
+
+    private void connectToServerButtonAction() {
+        if (isIPorPortNotPresent()) {
```

Comment:
I would consider inverting this, as the else block is a lot bigger than if block.

Reply:
fixed